engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterPlatformPluginTest.mm

```diff
@@ -417,9 +415,55 @@ - (void)testStatusBarHiddenUpdate {
                                         arguments:@[ @"SystemUiOverlay.top" ]];
   [plugin handleMethodCall:methodCallSet2 result:resultSet2];
   [self waitForExpectationsWithTimeout:1 handler:nil];
-#if not APPLICATION_EXTENSION_API_ONLY
   OCMVerify([mockApplication setStatusBarHidden:NO]);
-#endif
+
+  [flutterViewController deregisterNotifications];
+  [mockApplication stopMocking];
+  [bundleMock stopMocking];
+}
+
+- (void)testStatusBarHiddenNotUpdatedInAppExtension {
+  id bundleMock = OCMPartialMock([NSBundle mainBundle]);
+  OCMStub([bundleMock objectForInfoDictionaryKey:@"UIViewControllerBasedStatusBarAppearance"])
+      .andReturn(@NO);
+  OCMStub([bundleMock objectForInfoDictionaryKey:@"NSExtension"]).andReturn(@{
+    @"NSExtensionPointIdentifier" : @"com.apple.share-services"
+  });
+  id mockApplication = OCMClassMock([UIApplication class]);
+  OCMStub([mockApplication sharedApplication]).andReturn(mockApplication);
+  OCMReject([mockApplication setStatusBarHidden:OCMOCK_ANY]);
+
+  // Enabling system UI overlays to update status bar.
+  FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"test" project:nil];
+  [engine runWithEntrypoint:nil];
+  FlutterViewController* flutterViewController =
+      [[FlutterViewController alloc] initWithEngine:engine nibName:nil bundle:nil];
+
+  // Update to hidden.
+  FlutterPlatformPlugin* plugin = [engine platformPlugin];
+
+  XCTestExpectation* enableSystemUIOverlaysCalled =
+      [self expectationWithDescription:@"setEnabledSystemUIOverlays"];
+  FlutterResult resultSet = ^(id result) {
+    [enableSystemUIOverlaysCalled fulfill];
+  };
+  FlutterMethodCall* methodCallSet =
```

Comment:
Readability: maybe `systemOverlaysBottomExpectation`, `systemOverlaysBottomResult` and `setSystemOverlaysBottomCall`.